crates/common/mqtt_channel/src/config.rs

```diff
@@ -107,13 +110,19 @@ impl Default for AuthenticationConfig {
     }
 }
 
+#[derive(Debug, Clone)]
+enum ClientAuthConfig {
+    Cert(ClientAuthCertConfig),
+    User(ClientAuthUserConfig),
```

Comment:
Password is clearer.


```suggestion
    Pass(ClientAuthPassConfig),
```

Reply:
Ok good to know. I don't think we need to go to the level of having a "use_identity_as_username" equivalent in thin-edge.io...I'd much rather let the users configure the values themselves, and then possibly allow users to set a value using variables `${.common_name}` (but we can do template evaluation that later).